ingestion/tests/unit/readers/test_parquet_batches.py

```diff
@@ -0,0 +1,306 @@
+#  Copyright 2025 Collate
+#  Licensed under the Collate Community License, Version 1.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/LICENSE
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""
+Tests for ParquetDataFrameReader _read_parquet_in_batches method
+"""
+import unittest
+from unittest.mock import Mock, patch
+
+import pandas as pd
+
+from metadata.generated.schema.entity.services.connections.database.datalakeConnection import (
+    LocalConfig,
+)
+from metadata.readers.dataframe.parquet import ParquetDataFrameReader
+
+
+class TestParquetBatchReading(unittest.TestCase):
+    """
+    Test the _read_parquet_in_batches method functionality
+    """
+
+    def setUp(self):
+        """Set up test fixtures"""
+        self.config_source = LocalConfig()
+        self.reader = ParquetDataFrameReader(self.config_source, None)
+
+    def test_successful_batch_reading_with_iter_batches(self):
+        """Test successful batched reading when iter_batches is available"""
+        # Create mock parquet file with iter_batches capability
+        mock_parquet_file = Mock()
+        mock_parquet_file.iter_batches = Mock()
+
+        # Create sample data for two batches
+        batch1_data = pd.DataFrame(
+            {"id": [1, 2, 3], "name": ["Alice", "Bob", "Charlie"], "age": [25, 30, 35]}
+        )
+        batch2_data = pd.DataFrame(
+            {"id": [4, 5], "name": ["David", "Eve"], "age": [40, 45]}
+        )
+
+        # Create mock arrow table batches
+        mock_batch1 = Mock()
+        mock_batch1.to_pandas.return_value = batch1_data
+        mock_batch2 = Mock()
+        mock_batch2.to_pandas.return_value = batch2_data
+
+        mock_parquet_file.iter_batches.return_value = iter([mock_batch1, mock_batch2])
```

Comment:
good we're mocking the iter_batches, but would be even better to have a parquet file we can validate e2e. Even if its small, we might be OK limiting batches by smaller size and see it's handled properly?